src/browser/core_state.ts

```diff
@@ -862,10 +852,28 @@ export interface OfView extends Identity {
 }
 export function addBrowserView (opts: BrowserViewOpts, view: BrowserView) {
     const {uuid, name, target} = opts;
-    const ofView = { frames: new Map(), uuid, _options: opts, name, view, target, entityType: Shapes.EntityType.VIEW };
+    const ofView = {
+        frames: new Map(),
+        uuid,
+        _options: opts,
+        name,
+        view,
+        target,
+        entityType: Shapes.EntityType.VIEW,
+        preloadScripts: (opts.preloadScripts || []),
+        framePreloadScripts: {} // frame ID => [{url, state}]
+    };
     views.push(ofView);
     return ofView;
 }
+export function getEntityByUuidFrame(uuid: string, name: string): Shapes.InjectableContext | false {
+   const win = getWindowByUuidName(uuid, name);
+   if (win) {
+       return win;
+    } else {
+        return getBrowserViewByIdentity({uuid, name});
+    }
+}
```

Comment:
Does this function only apply for window and browserview entity types?

Reply:
Yes